charts/cla/templates/deployment.yaml

```diff
@@ -0,0 +1,82 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "cla.fullname" . }}
+  labels:
+    {{- include "cla.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "cla.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "cla.selectorLabels" . | nindent 8 }}
+    spec:
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      serviceAccountName: {{ include "cla.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+          - name: HOST
+            value: {{ .Values.claAssistant.host }}
+          - name: PORT
+            value: {{ .Values.claAssistant.port | default 80 }}
+          - name: HOST_PORT
+            value: {{ .Values.claAssistant.hostPort | default 8080 }}
+          - name: PROTOCOL
+            value: {{ .Values.claAssistant.protocol | default "http" }}
+          - name: GITHUB_CLIENT
+            value: {{ .Values.githubClient | default "CLIENT" }}
+          - name: GITHUB_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: github-credentials
```

Comment:
We should use a template partial for any value that is used in more than a file, tthe same as you do with the rest of templates in `_helpers.tpl`. Instead of the constant `github-credentials` we should use `{{ include "cla.githubCredentials" . }}`